Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,35 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
```

Comment:
You should be able to do something like:

```ruby
formula do
...
  def plist
    "..."
  end
end
```

It's just preferable to do more of an integration test using an actual formula class than a unit test where mocking is done. We want to avoid mocking whenever we can do so; it aids coverage and testing the code actually works interfacing with other code as it normally would.